### ISSUE TITLE
Add strategy for iRODS installation and setup

### ISSUE THREAD

**Issue by alanking** [irods/irods_testing_environment#12]
Need to implement the Strategy pattern with respect to iRODS installation and setup and add support for the following versions:
 - [ ] 4.3.x
 - [ ] 4.2.x (already implemented)
 - [ ] 4.1.x
 - [ ] 3.3.1 (and beyond?)

**Comment by trel** [comment]
3.3.1... but nothing older.

**Comment by trel** [comment]
4.1.12

**Comment by alanking** [comment]
Right, I think we might be able to get 4.1.x if it didn't change too much before then, but 3.3.1 might be the only one from before that which we may be able to support.

Also, every version of iRODS so far has put the server logs in a different location, so log collection in the run_tests.py script needs to implement a Strategy for each version there as well

**Comment by alanking** [comment]
Now that 4.3.x and 4.2.x are in the same branch, there will be a split in the setup logic as soon as irods/irods#6272 is merged. So, this issue will become very important imminently.
